src/geo/transform.js

```diff
@@ -1417,15 +1417,56 @@ class Transform {
         return new LngLatBounds(new LngLat(west, south), new LngLat(east, north));
     }
 
+    // Gets minimum bounding lngLats for non-Mercator and non-globe projections.
+    _getBoundsProjection(): LngLatBounds {
+        assert(!this.projection.supportsTerrain, "This function doesn't account for terrain");
```

Comment:
:+1: on these assertion. Could we add the same level of assertion in the other variants mentioned above since we have a lot now?